packages/wouter/src/use-hash-location.js

```diff
@@ -25,15 +25,25 @@ const currentHashLocation = () => "/" + location.hash.replace(/^#?\/?/, "");
 export const navigate = (to, { state = null } = {}) => {
   // calling `replaceState` allows us to set the history
   // state without creating an extra entry
+
+  let hash = to.replace(/^#?\/?/, "");
+  let search = location.search;
+
+  const searchIdx = hash.indexOf("?");
+  if (searchIdx !== -1) {
+    search = hash.slice(searchIdx, hash.length);
+    hash = hash.slice(0, searchIdx);
+  }
```

Comment:
I think this could be simplified to save up some bytes (haven't tested though)

```suggestion
  const [hash, search = location.search] = to.replace(/^#?\/?/, "").split("?")
```